### PR TITLE
MAINT: Unify polynomial valnd functions

### PR DESCRIPTION
**Pull request by eric-wieser**
No point writing the same function 12 times, when you can write it once.

Done using a find-and-replace, so should be free of typos in the individual polynomial files.

Inspired by #13078

### CODE REVIEW

File: numpy/polynomial/polyutils.py [hunk 1]
```diff
@@ -459,3 +459,51 @@ def _vander3d(vander_f, x, y, z, deg):
...
+
+    Parameters
+    ----------
+    val_f : function(array_like, array_like, tensor: bool) -> array_like
+        The ``<type>val`` function, such as ``polyval``
+    c, args :
+        See the ``<type>val<n>d`` functions for more detail
+    """
+    try:
+        args = tuple(np.array(args, copy=False))
```
Comment (eric-wieser) [inline numpy/polynomial/polyutils.py]: Future work - probably use `args = np.broadcast_arrays(*args)` here instead, now that it's in one place.

File: numpy/polynomial/polyutils.py [hunk 2]
```diff
@@ -459,3 +459,51 @@ def _vander3d(vander_f, x, y, z, deg):
...
+    Parameters
+    ----------
+    val_f : function(array_like, array_like, tensor: bool) -> array_like
+        The ``<type>val`` function, such as ``polyval``
+    c, args :
+        See the ``<type>grid<n>d`` functions for more detail
+    """
+    for xi in args:
+        c = val_f(xi, c)
+    return c
```
Comment (charris) [inline numpy/polynomial/polyutils.py]: Might as well put in the newline.

Comment (eric-wieser) [reply numpy/polynomial/polyutils.py]: Will update my editor config to not let me do this in future.

**Comment by charris** [comment]
LGTM, just the `no newline` nit. Feel free to put this in after fix.

**Review by charris — APPROVED** [review]

**Comment by eric-wieser** [comment]
Since this needs a fixup anyway, mind looking at #13078 one more time first? Then I can rebase this on top of the merged that.

File: numpy/polynomial/polyutils.py (hunk 1 above)
Comment (charris) [reply numpy/polynomial/polyutils.py]: Good thought.

**Comment by charris** [comment]
> mind looking at #13078 one more time first

Done and merged.

**Comment by eric-wieser** [comment]
Thanks. Will be interesting to see whether LGTM picks up on any of these changes whenever it next runs. Will rebase this one in the next few hours.

**Comment by eric-wieser** [comment]
Trailing newline removed, will merge once CI passes

**Pull request merged**